### PR TITLE
[AGW] pipelineD: configure SGi management ip address.

### LINKED ISSUE #2514: AGW: add static configuration for SGi interface.

### PR DESCRIPTION
**Pull request by pshelar**
## Summary

This allows operator to configure static IP address on
SGi for management traffic.
Only single IP is allowed.

Signed-off-by: Pravin B Shelar <pbshelar@fb.com>

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/python/magma/pipelined/main.py [hunk 1]
```diff
@@ -60,6 +60,8 @@ def main():
     logging.info("Nat: %s", enable_nat)
     vlan_tag = service.config.get('sgi_management_iface_vlan',
                                   service.mconfig.sgi_management_iface_vlan)
+    vlan_tag = service.config.get('sgi_management_iface_ip_addr',
```
Comment (koolzz) [inline lte/gateway/python/magma/pipelined/main.py]: did you mean `service.config['sgi_management_iface_vlan'] = ip_addr`

File: lte/gateway/python/magma/pipelined/tests/pipelined_test_util.py [hunk 2]
```diff
@@ -552,3 +553,26 @@ def get_ovsdb_port_tag(port_name: str) -> str:
...
+            tokens = str(port).strip().split()
+            port_name = tokens[0]
+            tag = tokens[1]
+            if port_name == port_name:
+                return str(tag).split('\\')[0]
+        except ValueError:
+            pass
+
+
+def get_iface_ipv4(iface: str):
```
Comment (koolzz) [inline lte/gateway/python/magma/pipelined/tests/pipelined_test_util.py]: Maybe annotate that it returns a list(I'm assuming its string)?
`-> List[string]`

File: lte/gateway/python/magma/pipelined/tests/pipelined_test_util.py [hunk 3]
```diff
@@ -552,3 +553,26 @@ def get_ovsdb_port_tag(port_name: str) -> str:
             pass
 
 
+def get_ovsdb_port_tag(port_name: str) -> str:
+    dump1 = subprocess.Popen(["ovsdb-client", "dump", "Port", "name", "tag"],
```
Comment (koolzz) [inline lte/gateway/python/magma/pipelined/tests/pipelined_test_util.py]: dump1->dump

File: lte/gateway/python/magma/pipelined/app/uplink_bridge.py [hunk 4]
```diff
@@ -170,3 +173,24 @@ def _del_eth_port(self):
...
             self.logger.debug("ignore port del error: %s ", ex)
+
+    def _set_sgi_static_ip(self):
+        if self.config.sgi_management_iface_ip_addr is None or \
+                self.config.sgi_management_iface_ip_addr == "":
+            return
+
+        try:
+            # Kill dhclient if running.
+            dump1 = subprocess.Popen(["pgrep", "-f", "dhclient.*" + self.config.uplink_bridge],
```
Comment (koolzz) [inline lte/gateway/python/magma/pipelined/app/uplink_bridge.py]: dump1->dump

File: lte/gateway/python/magma/pipelined/app/uplink_bridge.py [hunk 5]
```diff
@@ -170,3 +173,24 @@ def _del_eth_port(self):
...
+                                     stdout=subprocess.PIPE)
+            for pid in dump1.stdout.readlines():
+                subprocess.check_call(["kill", pid])
+            set_ip = ["ip",
```
Comment (koolzz) [inline lte/gateway/python/magma/pipelined/app/uplink_bridge.py]: set_ip->set_ip_cmd

**Review by koolzz — CHANGES_REQUESTED** [review]
Mostly nits

File: lte/gateway/python/magma/pipelined/tests/pipelined_test_util.py (hunk 3 above)
Comment (koolzz) [inline lte/gateway/python/magma/pipelined/tests/pipelined_test_util.py]: dump1->dump

**Review by koolzz — APPROVED** [review]
lgtm

File: lte/gateway/python/magma/pipelined/tests/pipelined_test_util.py [hunk 6]
```diff
@@ -552,3 +553,26 @@ def get_ovsdb_port_tag(port_name: str) -> str:
...
+                             stdout=subprocess.PIPE)
+    for port in dump1.stdout.readlines():
+        if port_name not in str(port):
+            continue
+        try:
+            tokens = str(port).strip().split()
+            port_name = tokens[0]
+            tag = tokens[1]
+            if port_name == port_name:
```
Comment (xjtian) [inline lte/gateway/python/magma/pipelined/tests/pipelined_test_util.py]: what's up with this? it's a tautology

**Review by xjtian — CHANGES_REQUESTED** [review]
one thing is fishy

File: lte/gateway/python/magma/pipelined/tests/pipelined_test_util.py (hunk 6 above)
Comment (pshelar) [reply lte/gateway/python/magma/pipelined/tests/pipelined_test_util.py]: nice catch.

**Review by xjtian — APPROVED** [review]

**Pull request merged**